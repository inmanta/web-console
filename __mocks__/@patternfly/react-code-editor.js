/* eslint-disable */
const React = require("react");

// Mock for @patternfly/react-code-editor
module.exports = {
  CodeEditor: ({
    code,
    onChange,
    language,
    height,
    isReadOnly,
    isDownloadEnabled,
    isCopyEnabled,
    isLanguageLabelVisible,
    customControls,
    isDarkTheme,
    ...props
  }) => {
    return (
      <div data-testid="code-editor" style={{ height }}>
        <pre data-testid="code-editor-content" {...props}>
          {code}
        </pre>
        {customControls && <div data-testid="code-editor-custom-controls">{customControls}</div>}
        {onChange && (
          <textarea
            data-testid="code-editor-textarea"
            value={code}
            onChange={(event) => onChange(event.target.value)}
            style={{ display: "none" }}
          />
        )}
      </div>
    );
  },
  CodeEditorControl: ({ onClick, icon, "aria-label": ariaLabel, tooltipProps, ...props }) => (
    <button
      onClick={onClick}
      aria-label={ariaLabel}
      data-tooltip={tooltipProps?.content}
      {...props}
    >
      {icon}
    </button>
  ),
  Language: {
    markdown: "markdown",
    json: "json",
    xml: "xml",
    python: "python",
    yaml: "yaml",
    javascript: "javascript",
    typescript: "typescript",
  },
};

/* eslint-enable */
