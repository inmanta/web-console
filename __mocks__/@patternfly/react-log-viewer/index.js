/* eslint-disable */

const React = require("react");

module.exports = {
  LogViewer: ({ data, toolbar, onScroll, hasLineNumbers = false, scrollToRow }) => {
    const handleScroll = (event) => {
      // Use the custom event detail if available, otherwise use default value
      const scrollOffsetToBottom = event?.detail?.scrollOffsetToBottom ?? 100;
      onScroll?.({ scrollOffsetToBottom });
    };

    return React.createElement(
      "div",
      {
        role: "log",
        "data-testid": "log-viewer",
        onScroll: handleScroll,
      },
      toolbar,
      React.createElement("pre", null, data)
    );
  },
  LogViewerSearch: ({ placeholder }) =>
    React.createElement("input", {
      type: "text",
      placeholder: placeholder,
    }),
};

/* eslint-enable */
