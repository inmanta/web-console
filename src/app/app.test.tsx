import React from 'react';
import { App } from '@app/index';
import { mount } from 'enzyme';
import { Button } from '@patternfly/react-core';
import Keycloak from 'keycloak-js';
import { wait } from '@testing-library/react';

describe('App tests', () => {
  fetchMock.mockResponse(JSON.stringify({}));
  let keycloak: Keycloak.KeycloakInstance;
  beforeEach(() => {
    keycloak = Keycloak();
  });

  it('should render a nav-toggle button', async () => {
    const wrapper = mount(<App keycloak={keycloak} shouldUseAuth={false} />);
    const button = wrapper.find(Button);
    await wait();
    expect(button.exists()).toBe(true);
  });

  it('should hide the sidebar when clicking the nav-toggle button', async () => {
    const wrapper = mount(<App keycloak={keycloak} shouldUseAuth={false} />);
    const button = wrapper.find('#nav-toggle').hostNodes();
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-expanded'));
    button.simulate('click');
    await wait();
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed'));
  });
});
