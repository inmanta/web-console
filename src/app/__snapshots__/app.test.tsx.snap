// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<t
  LoadingComponent={null}
  initConfig={
    Object {
      "flow": "implicit",
      "onLoad": "login-required",
    }
  }
  isLoadingCheck={null}
  keycloak={
    Keycloak {
      "accountManagement": [Function],
      "callback_id": 0,
      "clearToken": [Function],
      "createAccountUrl": [Function],
      "createLoginUrl": [Function],
      "createLogoutUrl": [Function],
      "createRegisterUrl": [Function],
      "hasRealmRole": [Function],
      "hasResourceRole": [Function],
      "init": [Function],
      "isTokenExpired": [Function],
      "loadUserInfo": [Function],
      "loadUserProfile": [Function],
      "login": [Function],
      "logout": [Function],
      "register": [Function],
      "updateToken": [Function],
    }
  }
  onError={null}
  onEvent={null}
  onToken={null}
>
  <BrowserRouter>
    <AppLayout>
      <AppRoutes />
    </AppLayout>
  </BrowserRouter>
</t>
`;
